src/Commands/Modules/NonEssentialModules.php

```diff
@@ -0,0 +1,238 @@
+            ->setDescription('Manage non-essential and stats modules.')
+            ->setHelp('This command Uninstall or Install non-essential and stats modules.')
+            ->addArgument(
+                'action',
+                InputArgument::OPTIONAL,
+                '( possible values : ' . $this->getPossibleActions() . ') ',
+                'status'
+            )
+            ->addOption(
+                'modulename',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Module to Install / Uninstall'
+            )
+            ->addOption(
+                'idsmodule',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'IDs module to Install or Uninstall separate by coma'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $moduleManager = $this->getContainer()->get('prestashop.module.manager');
+        $action = $input->getArgument('action');
+        $modulesInfos = [];
+        $modulesStatsInfos = [];
+
+        switch ($action) {
+            case 'status':
+                $io->text('<info>Non-essential Modules Informations</info>');
+
+                foreach (self::NON_ESSENTIAL_MODULES as $key => $uselessModule) {
+                    $modulesInfos[] = ['id' => $key, 'name' => $uselessModule, 'present' => $this->moduleExist($uselessModule) ? 'yes' : 'no', 'installed' => $moduleManager->isInstalled($uselessModule) ? 'yes' : 'no'];
+                }
+
+                $io->table(['ID', 'Name', 'Present?', 'Installed?'], $modulesInfos);
+                $io->text('You can'
+                    . PHP_EOL . '    - uninstall all modules by running  : `./bin/console fop:modules uninstall`'
+                    . PHP_EOL . '    - uninstall modules by running       : `./bin/console fop:modules uninstall --idsmodule x,y,z`'
+                    . PHP_EOL . '    - uninstall one module by running    : `./bin/console fop:modules uninstall --modulename ModuleName`'
+                    . PHP_EOL . '    - install all modules by running    : `./bin/console fop:modules install`'
+                    . PHP_EOL . '    - install modules by running         : `./bin/console fop:modules install --idsmodule x,y,z`'
+                    . PHP_EOL . '    - install one module by running     : `./bin/console fop:modules install --modulename ModuleName`');
+
+                return 0;
+            case 'uninstall':
+                $moduleToUninstall = $input->getOption('modulename');
+                $idsModule = $input->getOption('idsmodule');
+
+                if ($idsModule) {
+                    $idsModule = explode(',', $idsModule);
+                    foreach ($idsModule as $idModule) {
+                        if ($moduleManager->isInstalled(self::NON_ESSENTIAL_MODULES[$idModule])) {
```

Comment:
if user gives an id, why not uninstall it anyway ? regardless it's not essential or essential

Reply:
Because it's not the id_module it's the key of the array, i can change by the real id if you want ?

Reply:
non, je comprend mieux maintenant, c'est ok